### PR TITLE
feat: delete file operation

### PR DESCRIPTION
**Pull request by ypwong99**
This PR adds support for delete file operations and replaces #15.

### CODE REVIEW

File: remote_file_system/client_interface.py [hunk 1]
```diff
@@ -128,6 +130,22 @@ def write_file(self, file_path: Path, offset: int, number_of_bytes: int, content
...
+        incoming_message: DeleteFileResponse = send_message(
+            message=outgoing_message,
+            recipient_ip_address=self.server_ip_address,
+            recipient_port_number=self.server_port_number,
+            max_attempts_to_send_message=3,
+            timeout_in_seconds=5,
+        )
+        is_successful = incoming_message.is_successful
+        if is_successful is not True:
+            logger.error("Delete Failed. hehe")
```
Comment (ypwong99) [inline remote_file_system/client_interface.py:146]:
```suggestion
        if not is_successful:
            logger.error("Server responded that the delete file operation is not successful.")
```

File: remote_file_system/server_interface.py [hunk 2]
```diff
@@ -65,3 +65,11 @@ def subscribe_to_updates(
...
         self.subscribed_clients[relative_file_path].append(subscribed_client)
         return True
+
+    def delete_file(self, file_name: str) -> bool:
+        file_path = os.path.join(self.server_root_directory, file_name)
+        if not os.path.exists(file_path):
+            return False
+        else:
+            os.remove(file_path)
+            return True
```
Comment (ypwong99) [inline remote_file_system/server_interface.py:75]: We can consider removing the `else` keyword
```suggestion
        if not os.path.exists(file_path):
            return False
        os.remove(file_path)
        return True
```

File: tests/test_server_interface.py [hunk 3]
```diff
@@ -12,3 +13,12 @@ def test_read_file_file_exists():
...
         assert actual == expected
+
+    @staticmethod
+    def test_delete_file():
+        server_file_directory = os.getcwd()
+        f = open(f"{server_file_directory}/deleteme.txt", "w")
+        server = Server(server_root_directory=server_file_directory)
+        actual = server.delete_file("deleteme.txt")
+        expected = True
+        assert actual == expected
```
Comment (ypwong99) [inline tests/test_server_interface.py:24]:
```suggestion
        assert server.delete_file("deleteme.txt")
```

**Review by ypwong99 — COMMENTED** [review]
Hi @Eugene7997, thanks for the original PR #15, I rebased your branch on `main` and added comments here instead.

File: tests/test_server_interface.py [hunk 4]
```diff
@@ -12,3 +13,12 @@ def test_read_file_file_exists():
         actual = server.read_file("english_alphabets.txt")
         expected = "ABCDEFGHIJKLMNOPQRSTUVWXYZ".encode("UTF-8")
         assert actual == expected
+
+    @staticmethod
+    def test_delete_file():
+        server_file_directory = os.getcwd()
+        f = open(f"{server_file_directory}/deleteme.txt", "w")
```
Comment (ypwong99) [inline tests/test_server_interface.py:20]: We might want to close the file with something like
```
with open(f"{server_file_directory}/deleteme.txt", "w") as f:
    # Perform file operations
    pass
```

Comment (Eugene7997) [reply tests/test_server_interface.py:20]: sure. Alternatively f.close() would do as well but either is fine.

Comment (ypwong99) [reply tests/test_server_interface.py:20]: Sure, that works too

File: remote_file_system/client_interface.py (hunk 1 above)
Comment (Eugene7997) [reply remote_file_system/client_interface.py:146]: Looks good.

File: remote_file_system/server_interface.py (hunk 2 above)
Comment (Eugene7997) [reply remote_file_system/server_interface.py:75]: sure. either works for me

File: tests/test_server_interface.py (hunk 3 above)
Comment (Eugene7997) [reply tests/test_server_interface.py:24]: Thanks for heads up

**Pull request merged**